### ISSUE TITLE
spring-cloud-starter-alibaba-sentinel 2.2.1.RELEASE does not support Nacos auth

### ISSUE THREAD

**Issue by lude008** [alibaba/Sentinel#1618]
Nacos1.3.1，当nacos配置文件application.properties设置
nacos.core.auth.enabled=true时，此时sentinel-datasource的配置项缺少username、password，从而导致无法读取nacos的配置信息，报403错误.

**Comment by cdfive** [comment]
May use the second constructor of `NacosDataSource`:
`public NacosDataSource(final Properties properties, final String groupId, final String dataId, Converter<String, T> parser)`

```java
properties.put(PropertyKeyConst.USERNAME, "xxx");
properties.put(PropertyKeyConst.PASSWORD, "xxx");
```

Also refer to #1586, #272

**Comment by lude008** [comment]
> May use the second constructor of `NacosDataSource`:
> `public NacosDataSource(final Properties properties, final String groupId, final String dataId, Converter<String, T> parser)`
> 
> ```java
> properties.put(PropertyKeyConst.USERNAME, "xxx");
> properties.put(PropertyKeyConst.PASSWORD, "xxx");
> ```
> 
> Also refer to #1586, #272

你说的这个是在代码层的实现，但实际上当引用这边想要依赖于sentinel-datasource-nacos时其封装类无法实现，只能自己写代码实现。因为其对应的NacosDataSourceProperties和NacosDataSourceFactoryBean如下：
```
public class NacosDataSourceProperties extends AbstractDataSourceProperties {

	private String serverAddr;

	@NotEmpty
	private String groupId = "DEFAULT_GROUP";

	@NotEmpty
	private String dataId;

	private String endpoint;

	private String namespace;

	private String accessKey;

	private String secretKey;

	public NacosDataSourceProperties() {
		super(NacosDataSourceFactoryBean.class.getName());
	}

	@Override
	public void preCheck(String dataSourceName) {
		if (StringUtils.isEmpty(serverAddr)) {
			serverAddr = this.getEnv().getProperty(
					"spring.cloud.sentinel.datasource.nacos.server-addr",
					"localhost:8848");
		}
	}
....
}

// NacosDataSourceFactoryBean 类中的Properties 只是使用ip活云服务的accessKey、secretKey

public class NacosDataSourceFactoryBean implements FactoryBean<NacosDataSource> {

	private String serverAddr;

	private String groupId;

	private String dataId;

	private Converter converter;

	private String endpoint;

	private String namespace;

	private String accessKey;

	private String secretKey;

	@Override
	public NacosDataSource getObject() throws Exception {
		Properties properties = new Properties();
		if (!StringUtils.isEmpty(this.serverAddr)) {
			properties.setProperty(PropertyKeyConst.SERVER_ADDR, this.serverAddr);
		}
		else {
			properties.setProperty(PropertyKeyConst.ACCESS_KEY, this.accessKey);
			properties.setProperty(PropertyKeyConst.SECRET_KEY, this.secretKey);
			properties.setProperty(PropertyKeyConst.ENDPOINT, this.endpoint);
		}
		if (!StringUtils.isEmpty(this.namespace)) {
			properties.setProperty(PropertyKeyConst.NAMESPACE, this.namespace);
		}
		return new NacosDataSource(properties, groupId, dataId, converter);
	}
 ...
}
```

**Comment by cdfive** [comment]
May refer to 
[https://github.com/alibaba/spring-cloud-alibaba/pull/1354](https://github.com/alibaba/spring-cloud-alibaba/pull/1354)
[https://github.com/alibaba/spring-cloud-alibaba/issues/1350](https://github.com/alibaba/spring-cloud-alibaba/issues/1350) 
[https://github.com/alibaba/spring-cloud-alibaba/issues/1421](https://github.com/alibaba/spring-cloud-alibaba/issues/1421) 
The latest version of `spring-cloud-starter-alibaba-sentinel` has resovled this.

**Comment by lude008** [comment]
Currently, the latest version of Spring-cloud-alibaba-Starters is 2.2.1. The problem has not been solved, and nacos configuration is OK
```
spring:
  application:
    name: xxx-yyy-service
  cloud:
    nacos:
      # username and password is work
      username: nacos
      password: nacos@password
      discovery:
        server-addr: localhost:8848
    sentinel:
      transport:
        dashboard: localhost:8080
        port: 8719
      datasource:
        ds1:
          nacos:
            server-addr:  localhost:8848
            data-id: sentinel
            data-type: json
            rule-type: flow
           # username and password is not work
            username: nacos
            password: nacos@password
```

**Comment by sczyh30** [comment]
> Currently, the latest version of Spring-cloud-alibaba-Starters is 2.2.1. The problem has not been solved, and nacos configuration is OK

https://github.com/alibaba/spring-cloud-alibaba/pull/1354 may be released with 2.2.2.RELEASE. As a temporary solution, you may use raw `NacosDataSource` instead.
